pkg/kubelet/kubelet_node_status.go

```diff
@@ -1081,8 +970,13 @@ func (kl *Kubelet) defaultNodeStatusFuncs() []func(*v1.Node) error {
 			return nil
 		}
 	}
+	// if cloud is not nil, we expect the cloud resource sync manager to exist
+	var nodeAddressesFunc func() ([]v1.NodeAddress, error)
+	if kl.cloud != nil {
```

Comment:
almost seems like cloudResourceSyncManager should be a sub-structure of cloud.  I.E. kl.cloud.resourceSyncManager.NodeStatusFunctions() returns a list of functions (initially just NodeAddresses)